### PR TITLE
Add linter for upgrade proposals yml files

### LINKED ISSUE #398: Add linter for upgrade_proposal file
Add a linter for linting upgrade_proposal file

### PR DESCRIPTION
**Pull request by jonathan-eq**
Added separate linter for liting upgrade proposals yml files from github actions workflow, before starting building release.
Resolves #398

Depends on #391

### CODE REVIEW

File: komodo/lint_upgrade_proposals.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    upgrade_proposals_package, upgrade_proposals_package_version
+                )
+                print(
+                    f"Found package: '{upgrade_proposals_package}' with version"
+                    f" {upgrade_proposals_package_version} in repository"
+                )
+            except KomodoException as e:
+                errors.append(e.error)
+        if errors:
+            raise SystemExit(" \n ".join(errors))
```
Comment (berland) [inline komodo/lint_upgrade_proposals.py:30]: whitespace before `\n` should not be needed. Is the next `space` giving an intended visual effect indenting the output?

File: komodo/lint_upgrade_proposals.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if releases_with_upgrades_counter == 0:
+        raise ValueError("No upgrades found")
+    if releases_with_upgrades_counter > 1:
+        raise ValueError("Found upgrades for more than one release")
```
Comment (berland) [inline komodo/lint_upgrade_proposals.py:34]: is this an error condition? 🤔

Comment (berland) [reply komodo/lint_upgrade_proposals.py:34]: And why mixing `SystemExit` and `ValueError`?

File: komodo/lint_upgrade_proposals.py (hunk 1 above)
Comment (jonathan-eq) [reply komodo/lint_upgrade_proposals.py:30]: I do not remember why I added it, probably for some visual reasons. Removing it now.

File: komodo/lint_upgrade_proposals.py (hunk 2 above)
Comment (jonathan-eq) [reply komodo/lint_upgrade_proposals.py:34]: It is not an error, so it should be systemexit instead of valueerror.

**Review by berland — APPROVED** [review]
Looks good!

**Pull request merged**